interview-questions.md

```diff
@@ -8,36 +8,76 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-Researched answer:
+Researched answer: Params are information that is sent from the client to the server to gain ascertain exactly what the client wishes to do. As stated before, show, edit, update, destroy and create all require params. This is because their operation is soley based on what the client sends to the server. If the client wanted to see index, no params are necessary since all the info in the database will be shown to them. If show were to be called upon, a param is necessary since the client is requesting a specific piece of information.
 
 3. Name three rails generator commands. What is created by each?
 
-Your answer:
-
-Researched answer:
+Your answer: They are rails g model, rails g controller, and rails g migration. Model is used to create a table in the database where instances are made within the table following the attributes that are dictated by the model. Controller is used to create a class in which methods can be defined to function from requests made by users. Migration is used to add changes to already made tables within the database. An example of a model is creating one called car, where the attributes are wheels, make, and model. For the attributes, we would define what datatypes are taken such as wheels with integer, make with string, and model with string. Once the model is made we can create an instance allowing the empty table to be full of different instances of cars. An example of migration is forgetting to add the number of doors the car has so we can do the migration of doors with the datatype integer. The corresponding table would be car and we would do add column in the migration to show what we are doing to the car table. Then migrate command to conduct the changes.
+ 
+Researched answer: Generator commands allow developers to save a lot of time since it writes the boilerplate code necessary for the app to work for us. There are more generator commands than just the controller, model, and migration. They able to be found in generator gems that can be found from the web. They all have their uses. Such as scaffold, it is able to create the models, views, and controllers all by itself. Other commands are assets, channel, helper, integration_test, jbuilder, etc.
 
 4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
 
 action: "GET" location: /students
-
+index
```

Comment:
You did a great job naming the controllers relative to the routes, but make sure to add additional context to describe what each one is doing.  For instance, what is the difference between new and create?